packages/root-cms/ui/components/DocEditor/fields/DateTimeField.tsx

```diff
@@ -1,55 +1,126 @@
 import {FieldProps} from './FieldProps.js';
 
 export function DateTimeField(props: FieldProps) {
+  const field = props.field as DateTimeFieldSchema;
   const [value, setValue] = useDraftDocValue<Timestamp | null>(props.deepKey);
 
+  // Metadata is stored in a sibling field prefixed with `@`.
+  // e.g. if field is `scheduledAt`, metadata is in `@scheduledAt`.
+  // Only listen to the timezone key specifically.
+  const timezoneKey = `${getMetadataKey(props.deepKey)}.timezone`;
+  const [timezone, setTimezone] = useDraftDocValue<string | null>(timezoneKey);
+
+  const activeTimezone =
+    field.timezone ||
+    timezone ||
+    Intl.DateTimeFormat().resolvedOptions().timeZone;
+
   const dateStr = useMemo(() => {
     if (!value) {
       return '';
     }
-    return toDateStr(value);
-  }, [value]);
+    // value is Timestamp. toDate() gives a Date object (UTC).
+    // We want to format it in the selected timezone.
+    try {
+      return formatInTimeZone(
+        value.toDate(),
+        activeTimezone,
+        "yyyy-MM-dd'T'HH:mm"
+      );
+    } catch (err) {
+      console.error('Error formatting date:', err);
+      return '';
+    }
+  }, [value, activeTimezone]);
+
+  const [error, setError] = useState<string | null>(null);
+
+  const onDateChange = (newDateStr: string, validity?: ValidityState) => {
+    if (validity && !validity.valid) {
+      console.warn('Invalid date input');
+      setError('Invalid datetime');
+      return;
+    }
 
-  const onChange = (newDateStr: string) => {
     if (newDateStr) {
-      const millis = Math.floor(new Date(newDateStr).getTime());
-      const newValue = Timestamp.fromMillis(millis);
-      setValue(newValue);
+      // newDateStr is "YYYY-MM-DDTHH:mm"
+      // We interpret this as being in `timezone`.
+      try {
+        const date = fromZonedTime(newDateStr, activeTimezone);
+        // Check if date is valid
+        if (isNaN(date.getTime())) {
+          console.warn('Invalid date:', newDateStr);
+          setError('Invalid datetime');
+          return;
+        }
+        setError(null);
+        setValue(Timestamp.fromDate(date));
+      } catch (err) {
+        console.error('Error parsing date:', err);
+        setError('Invalid datetime');
+      }
     } else {
+      setError(null);
       setValue(null);
     }
   };
 
+  const onTimezoneChange = (newTimezone: string) => {
+    setTimezone(newTimezone);
+  };
+
   return (
     <div className="DocEditor__DateTimeField">
-      <input
-        type="datetime-local"
-        value={dateStr}
-        onChange={(e: Event) => {
-          const target = e.target as HTMLInputElement;
-          const newDateStr = target.value;
-          onChange(newDateStr);
-        }}
-      />
-      <div className="DocEditor__DateTimeField__timezone">
-        timezone: {Intl.DateTimeFormat().resolvedOptions().timeZone}
+      <div style={{display: 'flex', gap: '8px', alignItems: 'center'}}>
```

Comment:
Can you move these inline styles to css?

Reply:
done